lib/td/client/api/job.rb

```diff
@@ -287,6 +248,114 @@ def query(q, type=:hive, db=nil, result_url=nil, priority=nil, retry_limit=nil,
 
   private
 
+  def validate_content_length(response, current_total_chunk_size)
+    if expected_size = response.header['Content-Range'][0]
+      expected_size = expected_size[/\d+$/].to_i
+    elsif expected_size = response.header['Content-Length'][0]
+      expected_size = expected_size.to_i
+    end
+
+    if expected_size.nil?
+    elsif current_total_chunk_size < expected_size
+      # too small
+      raise EOFError, 'httpclient IncompleteError'
+    elsif current_total_chunk_size < expected_size
+      # too large
+      raise_error("Get job result failed", res)
```

Comment:
I'm just wondering why `else` call `raise_error`. `raise_error` will raise HTTP status code conditional error.
In the "too small" situation, is assumed as incomplete/EOF error. It is reasonable to me.
But "too large" situation is mysterious. I can't imagine such situation.. but at least I think it is not plain HTTP error.